src/protobuf-net.Reflection/NameNormalizer.cs

```diff
@@ -165,7 +176,13 @@ public virtual string GetName(EnumValueDescriptorProto definition)
         {
             var name = definition?.Options?.GetOptions()?.Name;
             if (!string.IsNullOrWhiteSpace(name)) return name;
-            return AutoCapitalize(definition.Name);
+
+            if (CustomNamesConversion != null && CustomNamesConversion.TryGetValue(definition.Name, out var customName))
+                return customName;
+            else if (definition.Name.StartsWith($"{definition.Parent.Name}_"))
```

Comment:
this is fine "as is" - but: we should make sure it is running through the *language specific* `Escape` method; and in particular, we should change that `Escape` method so that if the first character isn't valid for a name, we should add a `_` prefix. In reality, this means that the C# and VB generator's `Escape` method should just check "is the first character a number?" - and if so, prefix with `_`.